modules/experimental-layers/src/gpu-screen-grid-layer/gpu-screen-grid-layer.js

```diff
@@ -98,12 +93,27 @@ export default class GPUScreenGridLayer extends Layer {
     if (changeFlags) {
       this._updateAggregation(changeFlags);
     }
+
+    this._updateColorUniforms(opts);
   }
 
   draw({uniforms}) {
-    const {minColor, maxColor, parameters = {}} = this.props;
-    const {model, maxCountBuffer, cellScale} = this.state;
-    uniforms = Object.assign({}, uniforms, {minColor, maxColor, cellScale});
+    const {parameters = {}} = this.props;
+    const minColor = this.props.minColor || DEFAULT_MINCOLOR;
+    const maxColor = this.props.maxColor || DEFAULT_MAXCOLOR;
+
+    // If colorDomain not specified we use default domain [1, maxCount]
+    // maxCount value will be deduced from aggregated buffer in the vertex shader.
+    const colorDomain = this.props.colorDomain || [1, 0];
+    const {model, maxCountBuffer, cellScale, shouldUseMinMax, colorRange} = this.state;
+    uniforms = Object.assign({}, uniforms, {
+      minColor,
```

Comment:
linear interpolation of color should be enough for now. But most of the cases, kepler.gl has an array of colors predefined to be used as color scale. In this case, maybe define a `matrix` of colors and a uniform for `step` is more useful.

Reply:
Yes, I want to address this separately have an issue to track it #1946, this PR I want to make GPUScreenGrid on par with existing ScreenGrid layer for official release in 6.0.